### PR TITLE
Weather app api review

### PR DESCRIPTION
**Pull request by rOluochKe**
### I consumed weather api to build a web app that has the following functionalities:
- Beautiful weather app Web api
- Vanilla Javascript
- Fetch, async, await
- Consuming api from weather api
- Form validation
- Bootstrap user-friendly interface

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/* eslint-disable prefer-destructuring */
+const form = document.querySelector('.top-banner form');
+const input = document.querySelector('.top-banner input');
+const msg = document.querySelector('.top-banner .msg');
+const list = document.querySelector('.ajax-section .cities');
```
Comment (bellom) [inline src/index.js:5]: Let's separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        </h2>
+        <div class="city-temp">${Math.round(main.temp)}<sup>°C</sup></div>
+        <figure>
+          <img class="city-icon" src="${icon}" alt="${weather[0].description}">
+          <figcaption>${weather[0].description}</figcaption>
+        </figure>
+      `;
+      li.innerHTML = markup;
+      list.appendChild(li);
+    })
```
Comment (bellom) [inline src/index.js:69]: Let's separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${inputVal}&appid=${apiKey}&units=metric`;
+
+  fetch(url)
+    .then((response) => response.json())
+    .then((data) => {
+      const {
+        main, name, sys, weather,
+      } = data;
+      const icon = `https://s3-us-west-2.amazonaws.com/s.cdpn.io/162656/${weather[0].icon}.svg`;
```
Comment (bellom) [inline src/index.js:52]: we need to make different requests for the Celsius data and the Fahrenheit data.
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      const markup = `
+        <h2 class="city-name" data-name="${name},${sys.country}">
+          <span>${name}</span>
+          <sup>${sys.country}</sup>
+        </h2>
+        <div class="city-temp">${Math.round(main.temp)}<sup>°C</sup></div>
+        <figure>
+          <img class="city-icon" src="${icon}" alt="${weather[0].description}">
+          <figcaption>${weather[0].description}</figcaption>
+        </figure>
```
Comment (bellom) [inline src/index.js:65]: **(Note: we need to toggle in-between Celsius or Fahrenheit )**
Your app should have a toggle control to determine if the result will be shown in Celsius or Fahrenheit (According to the Odin instructions, the screenshot below;)
image

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        main, name, sys, weather,
+      } = data;
+      const icon = `https://s3-us-west-2.amazonaws.com/s.cdpn.io/162656/${weather[0].icon}.svg`;
+
+      renderResult(main, name, sys, weather, icon);
+    })
+    .catch(() => {
+      msg.textContent = 'Please search for a valid city 😩';
+    });
+
```
Comment (shloch) [inline src/index.js:61]: this to me sounds like the only logical code here..the rest is DOM manipulation

**Pull request merged**